projects/self-creation-portal/src/app/services/form/form.service.ts

```diff
@@ -89,7 +90,36 @@ export class FormService {
         reject(error);
       }
     });
-  
-}
 
+  }
+
+  private createUrlWithParams(baseUrl: string, params: { [key: string]: any }): string {
+    let httpParams = new HttpParams();
+    for (const key in params) {
+      if (params.hasOwnProperty(key) && params[key] !== undefined && params[key] !== '') {
+        httpParams = httpParams.set(key, params[key]);
+      }
+    }
+    return `${baseUrl}?${httpParams.toString()}`;
+  }
+
+  private generateParams(pagination: any, filters: any, sortOptions: any): { [key: string]: any } {
```

Comment:
Please move the functions that related to other API calls to common service, if the file isn't there please create in shared module and use it.